Library/Homebrew/global.rb

```diff
@@ -19,6 +20,8 @@
 
 require "config"
 
+HOMEBREW_REPOSITORY.extend(GitRepositoryExtension)
```

Comment:
I agree and thus intentionally limited this to the very few `Pathname` instances that actually need this. That all methods are prefixed with `git` also helps to avoid possible conflicts.